easybuild/tools/github.py

```diff
@@ -395,6 +408,279 @@ def post_comment_in_issue(issue, txt, repo=GITHUB_EASYCONFIGS_REPO, github_user=
+    # salt to use names of remotes/branches that are created
+    salt = ''.join(random.choice(string.letters) for _ in range(5))
+
+    # we need files to create the PR with
+    if paths:
+        non_existing_paths = []
+        for path in paths:
+            if not os.path.exists(path):
+                non_existing_paths.append(path)
+
+        if non_existing_paths:
+            raise EasyBuildError("One or more non-existing paths specified: %s", ', '.join(non_existing_paths))
+    else:
+        raise EasyBuildError("No paths specified")
+
+    tmp_git_working_dir = tempfile.mkdtemp(prefix='git-working-dir')
+    os.rmdir(tmp_git_working_dir)
+
+    # copy or init git working directory
+    pr_target_repo = build_option('pr_target_repo')
+    git_working_dirs_path = build_option('git_working_dirs_path')
+    if build_option('git_working_dirs_path'):
+        workdir = os.path.join(git_working_dirs_path, pr_target_repo)
+        if os.path.exists(workdir):
+            try:
+                print_msg("copying %s..." % workdir)
+                shutil.copytree(workdir, tmp_git_working_dir)
+            except OSError as err:
+                raise EasyBuildError("Failed to copy git working dir %s to %s: %s", workdir, tmp_git_working_dir, err)
+
+    if not os.path.exists(tmp_git_working_dir):
+        mkdir(tmp_git_working_dir, parents=True)
+
+    git_repo = git.Repo.init(tmp_git_working_dir)
+
+    _log.debug("temporary git working directory ready at %s", tmp_git_working_dir)
+
+    if pr_target_repo != GITHUB_EASYCONFIGS_REPO:
+        raise EasyBuildError("Don't know how to create/update a pull request to the %s repository", pr_target_repo)
+
+    # add remote to pull from
+    github_url = 'https://github.com/%s/%s.git' % (target_account, pr_target_repo)
+    _log.debug("Cloning from %s", github_url)
+
+    origin = git_repo.create_remote('pr_target_account_%s_%s' % (target_account, salt), github_url)
+    if not origin.exists():
+        raise EasyBuildError("%s does not exist?", github_url)
+
+    if start_branch is None:
+        start_branch = build_option('pr_target_branch')
+
+    # git fetch
+    # can't use --depth to only fetch a shallow copy, since pushing to another repo from a shallow copy doesn't work
+    print_msg("fetching branch '%s' from %s..." % (start_branch, github_url))
+    try:
+        res = origin.fetch(start_branch)
+    except GitCommandError as err:
+        raise EasyBuildError("Failed to fetch branch '%s' from %s: %s", start_branch, github_url, err)
+    if res:
+        if res[0].flags & res[0].ERROR:
+            raise EasyBuildError("Fetching branch '%s' from remote %s failed: %s", start_branch, origin, res[0].note)
+        else:
+            _log.debug("Fetched branch '%s' from remote %s (note: %s)", start_branch, origin, res[0].note)
+    else:
+        raise EasyBuildError("Fetching branch '%s' from remote %s failed: empty result", start_branch, origin)
+
+    # git checkout -b <branch>; git pull
```

Comment:
yeah, well, I'm not sure yet...

The test is failing on Jenkins, and I'm not sure why. I'm just checking now if the `pull` makes a difference